test/hotspot/jtreg/compiler/c1/ComplexLockingAndMultiThreading.jasm

```diff
@@ -0,0 +1,194 @@
+public super class ComplexLockingAndMultiThreading
+  version 52:0
+{
+  private static final Field lock:"Ljava/lang/Object;";
+
+  public Method "<init>":"()V"
+    stack  1 locals  1
+  {
+         aload_0;
+         invokespecial     Method java/lang/Object."<init>":"()V";
+         return;
+  }
+  public static Method main:"([Ljava/lang/String;)V"
```

Comment:
There's no need to have the main method in jasm, right? I think it's only `synchronizedMethod` which triggers the issue. So ideally, the jasm file is as compact as possible to ease future maintenance of the test.

Reply:
IMO we even don't need to run the code to trigger compilation. We can use WhiteBox to compile it.